geo/src/algorithm/relate/geomgraph/geometry_graph.rs

```diff
@@ -336,6 +336,7 @@ where
 
         // determine the current location for the point (if any)
         let boundary_count = {
+            #[allow(clippy::bool_to_int_with_if)]
             let prev_boundary_count =
                 if Some(CoordPos::OnBoundary) == label.position(arg_index, Direction::On) {
```

Comment:
The clippy suggestion was to do something like: 

`let prev_boundary_count = usize::from(Some(CoordPos::OnBoundary) == label.position(arg_index, Direction::On))`

Which, in my opinion, is just too dense. 

Pulling out a temp could address that:
```
let is_on_boundary = Some(CoordPos::OnBoundary) == label.position(arg_index, Direction::On);
let prev_boundary_count = usize::from(is_on_boundary);
```

But, I still prefer what's already there. I guess my (perhaps controversial) opinion is that booleans shouldn't be treated as numbers and that clippy is wrong to encourage this. 🤷

Reply:
I'd write it as `matches!(label.position(arg_index, Direction::On), Some(CoordPos::OnBoundary)) as usize`, but yeah, whatever.

But if this comes up again, I'd be for globally suppressing this lint.